### ISSUE TITLE
Error on downloading with empty hashfile

### ISSUE THREAD

**Issue by Ymiros0** [nobbyfix/AzurLane-AssetDownloader#2]
Trying to download the assets caused an uncaught exception, seems like it's not handling the empty hashfile well if I had to take a guess?
The script output:

```[...]
BGM: Current version 1 is older than latest version 22.
Server returned empty hashfile for BGM, skipping.
Traceback (most recent call last):
  File "D:\Assets2\src\main.py", line 51, in <module>
    main()
  File "D:\Assets2\src\main.py", line 48, in main
    execute(args)
  File "D:\Assets2\src\main.py", line 28, in execute
    update_assets = updater.update(vresult, clientconfig.cdnurl, userconfig, CLIENT_ASSET_DIR, args.force_refresh)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Assets2\src\lib\updater.py", line 129, in update
    return _update(version_result, cdnurl, userconfig, client_directory)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Assets2\src\lib\updater.py", line 123, in _update
    return _update_from_hashes(version_result, cdnurl, userconfig, client_directory, oldhashes, newhashes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Assets2\src\lib\updater.py", line 114, in _update_from_hashes
    comparison_results = compare_hashes(oldhashes, newhashes)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Assets2\src\lib\updater.py", line 36, in compare_hashes
    results = {row.filepath: CompareResult(None, row, CompareType.New) for row in newhashes}
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable```

**Comment by nobbyfix** [comment]
fixed with 1b780ce36a81e78f200e1831129b8b4e777b3328

**Comment by nobbyfix** [comment]
also thank god it crashed, if i'd returned an empty list instead of None it would've chugged along and assumed all files need to be deleted
